src/GafferScene/CopyPrimitiveVariables.cpp

```diff
@@ -0,0 +1,208 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2019, John Haddon. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferScene/CopyPrimitiveVariables.h"
+
+#include "GafferScene/SceneAlgo.h"
+
+#include "Gaffer/ArrayPlug.h"
+
+#include "IECoreScene/Primitive.h"
+
+using namespace std;
+using namespace IECore;
+using namespace IECoreScene;
+using namespace Gaffer;
+using namespace GafferScene;
+
+GAFFER_GRAPHCOMPONENT_DEFINE_TYPE( CopyPrimitiveVariables );
+
+size_t CopyPrimitiveVariables::g_firstPlugIndex = 0;
+
+CopyPrimitiveVariables::CopyPrimitiveVariables( const std::string &name )
+	:	Deformer( name, 2, 2 )
```

Comment:
Is this approach the preferred new standard (as opposed to say Parent which has `in` and `child` plugs)? I found it confusing both in the code and in the gui to be honest.... In the code I was confused why you're grabbing `inPlugs()->getChild( 1 )` (whereas a named plug would be much more clear) and in the gui I was confused why the in0 and in1 nodules are named like its an array, but its not visualy the same as a normal arrayplug (they're spread apart more), and theres no labels or description to clarify the meaning...

Reply:
It's something I've been wondering about, and I think I got wrong with CopyAttributes and copied here. @themissingcow and I chatted about it a bit today and came up with the following :

https://github.com/GafferHQ/gaffer/wiki/DeveloperGuidelines#arrayplugs-vs-individual-inputs

Sound OK?

Reply:
Yep, that sounds like a good rule of thumb. Though, be ware, the wiki will need clarifying when we make Parent.child an arrayplug.